### PR TITLE
🐍 Remove lock on gmpy2 to allow python 3.10

### LINKED ISSUE #527: 🐍 Support Python 3.10 by unlocking version of gmpy2
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Actually the version of the gmpy2 package required by electionguard-python (2.0.8) is not compatible with Python 3.10 (see https://github.com/aleaxit/gmpy/pull/297 )

This error prevent from building wheels for gmpy2
`  ERROR: Failed building wheel for gmpy2`

### Expected Behavior

_No response_

### Steps To Reproduce

Try to install electionguard-python in a python 3.10 env.

### Environment

```markdown
-OS: Archlinux
-Python version: 3.10
```

### Anything else?

Why is 2.0.8 required ? Is there another (newer) version that could work or should I just downgrade the python version?

I think it's better to always target newer python versions, in addition that latests versions of gmpy includes binary wheels for various platforms.

I know that the README says that python 3.9 is *required* but I'm able to run this package with python 3.10 and gmpy2==2.1.2 without any problem (all tests pass) on up-to-date Archlinux so I thought it was required at the time it was written, but the version is fixed by `pyproject.toml`

### PR DESCRIPTION
**Pull request by keithrfung**
[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue
*Link your PR to an issue*

Fixes #527 

### Description
This pull request is removing a lock we have on gmpy2 that essentially locks the version due to math issues we had previously. This will loosen this up to allow python 3.10 to be used. The documentation has been updated surrounding the windows build since this is the piece directly affected. 

- Update Build and Run docs to give more options for installing gmpy2 for Windows
- Update poetry.toml and lock file to indicate change
- Remove previous binaries to shrink repo size
- Add admonitions to mkdocs

### Testing
`make serve` to check docs
`make` and `make test` to run existing test suite on Python 3.10

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -31,10 +31,7 @@ packages = [{ include = "electionguard", from = "src" }]
 
 [tool.poetry.dependencies]
 python = "^3.9.5"
-gmpy2 = "2.0.8"
-# For Windows Builds
-# gmpy2 = { path = "./packages/gmpy2-2.0.8-cp39-cp39-win_amd64.whl" } # 64 bit
-# gmpy2 = { path = "./packages/gmpy2-2.0.8-cp39-cp39-win32.whl" } # 32 bit
+gmpy2 = "^2.0.8"
```
Comment (keithrfung) [inline pyproject.toml:34]: The key one liner for change. The rest is mostly documentation.

**Review by thommignot — APPROVED** [review]
Nice, thank you!

**Review by SteveMaier-IRT — APPROVED** [review]
Just a comment about version numbers.  Overall good change.

File: docs/Build_and_Run.md [hunk 2]
```diff
@@ -28,16 +28,31 @@ OR
...
+    1. Determine if 64-bit:
+        _The 32 vs 64 bit is based on your installed python version NOT your system._
+        This code snippet will read true for 64 bit.
+    ```py
+    python -c 'from sys import maxsize; print(maxsize > 2**32)'
+    ```
+    2. Download [pre-compiled binary](https://www.lfd.uci.edu/~gohlke/pythonlibs/#gmpy) into project folder
+    3. Within `pyproject.toml`, replace `gmpy2` reference with direct path to downloaded file.
+    ```py
+    gmpy2 = { path = "./packages/gmpy2-2.0.8-cp39-cp39-win_amd64.whl" }
```
[the remark this answers is not in the document]
Comment (keithrfung) [reply docs/Build_and_Run.md:53]: I was just trying to demonstrate any numbers. The downloads don't include the latest.

**Pull request merged**